src/layer/AbstractSentinelHubV3Layer.ts

```diff
@@ -487,9 +491,10 @@ export class AbstractSentinelHubV3Layer extends AbstractLayer {
     return res.data;
   }
 
-  public async updateLayerFromServiceIfNeeded(reqConfig: RequestConfiguration): Promise<void> {
+  public async updateLayerFromServiceIfNeeded(reqConfig?: RequestConfiguration): Promise<void> {
     const layerParams = await this.fetchLayerParamsFromSHServiceV3(reqConfig);
     this.legend = layerParams['legend'] ? layerParams['legend'] : null;
+    this.dataProduct = layerParams['dataProduct'] ? layerParams['dataProduct'] : null;
```

Comment:
This doesn't look like a correct place to update `dataProduct` to me... If nothing else, there should be no difference between `evalscript` and `dataProduct`, so if one is set (from the service), the other one should be too. I have checked where this is used but failed to see where this is needed, as it is updated in `getMap` anyway - is there a special case that I missed?

Reply:
Yes, in getMap gets updated but the check of the apiType happens before getMap is called in EOB3. So the idea is to updateLayer and then check supportApiType to decide whether WMS or Processing is used.

Reply:
Still, this doesn't seem like the proper way to do it... Let's discuss live how to best solve this on Monday.

Reply:
As discussed, we can reolve this, but I would suggest adding a comment:
```
this.legend = ...
// this is a hotfix for `supportsApiType()` not having enough information - should be fixed properly later:
this.dataProduct = ...
```

Fftr.

Reply:
Done. Thanks Anze